WPS Migration: Add notice

Mayisha opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:
- Added a warning notice informing the merchants about the upcoming migration related to PayPal standard.

<img width="1288" height="83" alt="Screenshot 2025-08-22 at 3 42 50 PM" src="https://github.com/user-attachments/assets/eb30b264-630c-497b-91b9-6fc68ef20e38" />

Closes PAYPAL-73

### How to test the changes in this Pull Request:
- Set the deprecated PayPal gateway feature flag.
```
wp option patch update woocommerce_paypal_settings _should_load 'yes'
```
- Remove the `API credentials` from **WooCommerce > Settings > Payments > PayPal Standard** page if you have them added.
- Load any page in your WooCommerce admin dashboard. You should see the PayPal migration notice.
- Click the dismiss and confirm that the notice is removed. It should not reappear on page refresh.

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

Mayisha commented on plugins/woocommerce/includes/gateways/paypal/includes/class-wc-gateway-paypal-notices.php:
```diff
@@ -0,0 +1,157 @@
+		$dismiss_url = wp_nonce_url( add_query_arg( 'paypal_tos_action', 'dismiss' ), 'paypal_tos_action' );
+
+		echo '<div class="notice notice-warning is-dismissible">';
+		echo '<p>By continuing to use PayPal Standard, you agree to our updated <a href="https://www.example.com" target="_blank">Terms of Service</a>.</p>';
```
ToDo.

annemirasol reviewed (commented):
We should add a check to only load the ToS notice when PayPal Standard is actually active/enabled.

naman03malhotra committed:
[Migrator]: Add support for AVIF Images (#60426)

* Add support for AVIF Images

* Added changelog

Aljullu committed:
Use WP core Template registration API (#60191)

* Use WP core templating logic for WC taxonomy templates

* Remove duplicate 'taxonomy_template_hierarchy' hook

* Update outdated comment

* Fix Products by Attribute template not falling back properly

* Update tests

* Update PHP unit tests

* Add missing strict_types declare

* Add changelog file

* Fix fall back template some times not being added to the hierarchy when it was the first match

* Add spaces in strict_types declarations

* Add guard against missing

* Typo

* Avoid redundant array_search() call

* Add missing PHP types

* Use get_stylesheet() instead of wp_get_theme()->get_stylesheet()

* Update plugins/woocommerce/src/Blocks/Templates/AbstractTemplateWithFallback.php

Co-authored-by: Karol Manijak <20098064+kmanijak@users.noreply.github.com>

* Use the WordPress core template registration API to register WooCommerce templates (#60234)

* Add changelog file

* Use the WordPress core template registration API to register WooCommerce templates

* Linting

* Update PHP test

* Remove no longer necessary logic that adds template data

* Prevent duplicate templates appearing in the editor

* Prevent warning if array starts at 1

* Rename method also in tests

* Revert "Remove no longer necessary logic that adds template data"

This reverts commit 4ea0f667961e2cff165757331056aba40c701fa6.

* Add theme data to PHP test

* Make sure compatibility layer isn't added twice

* Typo

* Run hooks on templates registered via the Template Registration API

* Add e2e tests that check template priority

* Fix templates not being created if there was a fallback template with the WooCommerce slug

* Reorder code

* Linting

* Remove no longer necessary checks

* Create TemplatesPage in tests

* Add link to issue

* Remove extra checks

* Optimize add_block_templates()

* Linting

* Remove unnecessary test

* Typo

* Update plugins/woocommerce/src/Blocks/BlockTemplatesController.php

Co-authored-by: Karol Manijak <20098064+kmanijak@users.noreply.github.com>

* Update e2e tests to use the theme template slug instead of the woocommerce slug (#60297)

* Update e2e tests to use the theme template slug instead of the woocommerce slug

* Add changelog file

* Fix some tests

* Make createTemplate() more reliable

* Simplify has_block_template function (#60326)

* Simplify has_block_template function

* Add changelog file

---------

Co-authored-by: Karol Manijak <20098064+kmanijak@users.noreply.github.com>

* Improve 'Editor: In taxonomies templates' Product Collection flaky test with the new Template registration API (#60372)

* Improve 'Editor: In taxonomies templates' Product Collection flaky test

* Add changelog file

* Implement CodeRabbit suggestions to Template registration API PR (#60369)

* Apply CodeRabbit suggestions

* Optimize remove_duplicate_customized_templates()

* Fix docblock

* Make sure we check for the deprecated slug in remove_templates_with_custom_alternative()

* Add mising use

* Fix test

* Fix template parts not being listed in the Site Editor after migrating to the Template registration API (#60370)

* Make sure template parts are listed in the site editor

* Make tests actually open the template part

* Add changelog file

* Make tests actually open the template part (II)

* Avoid using selectors to wait for search to finish in tests

---------

Co-authored-by: Karol Manijak <20098064+kmanijak@users.noreply.github.com>

jorgeatorres committed:
Update release bump workflow to use next version for WC class property (#60356)

Update versioning script to use NEXT_VERSION variable in release-bump-version workflow

louwie17 committed:
Optimize product analytics product query for deleted products (#60419)

* Optimize query to get name of deleted product

* Add changelog

* Remove error log

ralucaStan committed:
Fix missing space in including x in taxes text in the cart block (#60466)

* Fix spacing issue in Taxes item in the order summary

* Add changelog

yuliyan committed:
Add log level threshold for Email Editor (#60255)

* Add log level threshold for Email Editor

* Add changelog entries

* Fix link to logging docs

This reverts commit 86cad8be03d4a16ceedad22d6b2d01fe8d9649be.

albarin committed:
Update the release kickoff templates with monitoring time (#60457)

* Update the release kickoff templates with monitoring time

* Update link

* Improve message

yuliyan committed:
Email Editor: Filter unnecessary styles from iframe (#60354)

* Email Editor: Filter unnecessary styles from iframe

* Add changelog

* Rework allowed stylesheets logic and address coderabbit suggestions

* Fix typos and comments

* Add shared styles to the allowed iframe handles

* Remove confusing type assertion

mordeth committed:
[Cleanup] Remove deprecated legacy NOX payment selection screen (#60444)

* Remove deprecated legacy NOX payment selection screen

opr committed:
Add `WOOCOMMERCE_CART` constant when on `update_cart` AJAX actions (#60467)

mordeth committed:
[NOX] Resolve SCSS load order issues causing build warnings (modals.scss) (#60456)

* Resolve SCSS load order issues causing build warnings

dinhtungdu committed:
Fix: register product entity title (#60458)

dinhtungdu committed:
Fix: Limit loading legacy single product scripts (#60223)

* fix: remove always loading legacy gallery script for block themes

* fix: load gallery scripts for classic template on render

* fix: load product image gallery scripts when block is inused

* fix: enqueue legacy single product script on demand

* docs: explain the duplication

* Add changefile(s) from automation for the following project(s): woocommerce

* fix: only enqueue single-product.js for legacy product details block

* refactor: change approach, advoid branching in WC_Frontend_Scripts which is for classic themes

* chore: phpcs

* fix: always loading neccessary assets for Product Image Gallery when used

---------

Co-authored-by: github-actions <github-actions@github.com>

Sidsector9 committed:
a11y/W0010UP-18: Ensure screen reader announcements are not interrupted when removing items from the Mini-Cart drawer (#60256)

* fix screen reader when product is removed from mini cart

* Add changefile(s) from automation for the following project(s): woocommerce

* remove data-keep-visible attr on drawer close

* fix JS linting errors

* update the documentation

* Update plugins/woocommerce/client/blocks/assets/js/base/components/drawer/utils/aria-helper.ts

---------

Co-authored-by: github-actions <github-actions@github.com>
Co-authored-by: Albert Juhé Lluveras <aljullu@gmail.com>

helgatheviking committed:
Make add to cart with options block validation extensible (#60113)

* Make add to cart with options form validation dependent on the presence of validation errors. First pass only only includes updating validation for Variable Products.

* Update plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/variation-selector/frontend.ts

remove testState debug code

Co-authored-by: Albert Juhé Lluveras <aljullu@gmail.com>

* Update plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/variation-selector/frontend.ts

use early returns

Co-authored-by: Albert Juhé Lluveras <aljullu@gmail.com>

* Update plugins/woocommerce/src/Blocks/BlockTypes/AddToCartWithOptions/AddToCartWithOptions.php

add translator note

Co-authored-by: Albert Juhé Lluveras <aljullu@gmail.com>

* remove removeError until we have an established use case for it

* checking for duplicate errors is redundant as the store notices already handles that

* no need to redeclare addToCartWithOptions actions as we are currently in that extended store

* Follow up to 1adef05c8c233df101339470f5848eb64a9905c7 as errorMessages got deleted

* remove watch callback from VariationSelectorAttributeOptions.php. It's handled in VariationSelector.php.

* Change initial server-side state for isFormValid to handle grouped products and all types that has_options()

* split handleSubmit into handleSubmit for displaying notices and addToCart for adding to the cart. This allows different product types to modify how addToCart happens without needing to duplicate the validation and notice display.

* Move grouped addToCart and setQuantity to its own extended store.

* Add validation to grouped product selector.

* use universalLock constant

* add changelog

* Minor cleanups

* Linting

* CodeRabbit suggestions

---------

Co-authored-by: Albert Juhé Lluveras <aljullu@gmail.com>
Co-authored-by: Albert Juhé Lluveras <contact@albertjuhe.com>

Aljullu committed:
Fix iAPI based Mini-Cart not showing some values when window.wc.blocksCheckout is missing (#60482)

* Fix iAPI based Mini-Cart not showing some values when window.wc.blocksCheckout is missing

* Add changelog

kmanijak committed:
Fix Product Collection flaky test (#60486)

* Adjust the editor util so there's method to close global inserter

* Close global inserter after inserting Product Collection

* Add changelog

xristos3490 committed:
Introduce alpha version of Back in Stock Notifications (#59947)

* MIC: Create base infrastructure and DB tables for Back in Stock Notifications (#57062)

* Add base service and database schema

* Make linter happy

* Linter ignore

* Rename datastore class

* Add basic CRUD for notifications, metadata, and a helper for logs

* Make linter happy again

* Shorten table names, suffix dates with gmt, introduce dedicated meta store

* Fix docblock on reusable datastore

* Lint fixes

* Lint fixes

* Update service provider

* Lint fixes

* Lint fixes

* Lint fixes

* Lint fixes

* Fix dbDelta issues with capital letters and phpcs fixes

* Add specific query formats

* Remove object keyword refs

* Remove meta cache duplication in favor of WC_Data level caching

* Improvements in stock notifications CRUD and introduce WP_Error returns

* Docblock improvements

* Create the main controller and wire the data store

* Fix data stores for basic CRUD and add unit tests

* Add metadata handling and unit tests

* Lint fixes

* Organize unit tests and align create and modified dates

* Alias main controller init and make linter happy

* One more lint fix

* Imporvements and new db table indexes

* Clean up notification unit tests

* Add metadata unit tests

* Rename the event log helper

* Add a factory, unit tests, and move notifications tests

* Lint fixes

* Make linter happy

* Add a query method

* Create data store for logs

* Lint fixes

* Lint fixes and fix the init data store of logs

* Fix error in log params

* Use the main datastore for tests

* Use specified date modified value and some polishing

* Fix unit tests by applying changes on the object after saving the meta

* Lint fixes

* Setup timezone for tests to control output

* Avoid error logging in unit tests

* Fix timezones when reading data from DB

* Lint fixes

* Reorg the table names into getters in data stores

* Minor fixes on the scheme

* Move delete functions in each data store

* Lint fixes

* Fix unit tests

* Remove comment ignores

* Use assertCount

* Fix unit tests

* Add args and return types in activity log data store

* Added args and return types in notifications data store where applicable

* Fix the datastore docblock in activity logs tests

* Differentiate dates for better coverage

* Remove text based tests

* Check non-existance after deleting meta but not saved yet

* One more rename of note to payload

* Modify the database schema and fix the tests

* Make linter happy

* Add the id at the batch processing index

* Cleanup is_queued

* Add docblock details on notification statuses

* Add all param types for date-related setters

* Add tests for query types

* Improve performance on the query method and the model instantiation

* Make linter happy

* Fix the NotificationCancellationSource enum

* Add type int in factory

* Make NotificationStatus enums public

* Add types on Notification setters

* simplify db index statement

* Set the cancellation_source db size and use array_key_exists on update

* Protect from failed deletions

* Review on datastore

* Unit Tests FTW

* Improve deletion checkings

* Rename controller to StockNotifications

* Fix the main entrypoint

* Fix typo

* MIC: Introduce Email Templates for Back In Stock Notifications (#58044)

* Add base service and database schema

* Make linter happy

* Linter ignore

* Rename datastore class

* Add basic CRUD for notifications, metadata, and a helper for logs

* Make linter happy again

* Shorten table names, suffix dates with gmt, introduce dedicated meta store

* Fix docblock on reusable datastore

* Lint fixes

* Lint fixes

* Update service provider

* Lint fixes

* Lint fixes

* Lint fixes

* Lint fixes

* Fix dbDelta issues with capital letters and phpcs fixes

* Add specific query formats

* Remove object keyword refs

* Remove meta cache duplication in favor of WC_Data level caching

* Improvements in stock notifications CRUD and introduce WP_Error returns

* Docblock improvements

* Create the main controller and wire the data store

* Fix data stores for basic CRUD and add unit tests

* Add metadata handling and unit tests

* Lint fixes

* Organize unit tests and align create and modified dates

* Alias main controller init and make linter happy

* One more lint fix

* Imporvements and new db table indexes

* Clean up notification unit tests

* Add metadata unit tests

* Rename the event log helper

* Add a factory, unit tests, and move notifications tests

* Lint fixes

* Make linter happy

* Add a query method

* Create data store for logs

* Lint fixes

* Lint fixes and fix the init data store of logs

* Fix error in log params

* Use the main datastore for tests

* Use specified date modified value and some polishing

* Fix unit tests by applying changes on the object after saving the meta

* Lint fixes

* Setup timezone for tests to control output

* Avoid error logging in unit tests

* Fix timezones when reading data from DB

* Lint fixes

* Reorg the table names into getters in data stores

* Minor fixes on the scheme

* Move delete functions in each data store

* Lint fixes

* Fix unit tests

* Remove comment ignores

* Use assertCount

* Fix unit tests

* Add args and return types in activity log data store

* Added args and return types in notifications data store where applicable

* Fix the datastore docblock in activity logs tests

* Differentiate dates for better coverage

* Remove text based tests

* Check non-existance after deleting meta but not saved yet

* One more rename of note to payload

* Introduce the email templates

* Simplify and reorg templates and controllers

* Lint fixes

* Add plain templates

* Handle plain mode in template functions

* Lint fixes

* Lint fixes

* Reorg of files, renamed the email classes, and introduced a single entrypoint on the Controller class

* Lint fixes

* Lint fixes

* revert unrelated changes

* Rename email titles

* Modify the database schema and fix the tests

* Make linter happy

* Add the id at the batch processing index

* Fix into->intro type

* Rename custom tax location method

* Fix verify template

* Improve getter for formatted variations and add a unit test

* Add unit tests for get_product_permalink/name

* Rename emails id and classes for better semantics

* Lint fixes on tests

* Lint fixes on tests

* Reorg namespaces into Emails, and move the templates into the email controller

* Rename EmailManager, move two emails in transactional emails actions and modify trigger points

* Lint

* Fix customer data hook

* Renamed confirm email to verified, and added better docblock for the customer data hook

* Simplify and repurpose get_product_formatted_variation_list; handle emails rendering on the spot

* Rename the template

* Lint

* Cleanup renamed files

* Revert minor change on the provider

* MIC: Add Customer Stock Notifications settings (#58202)

* Add base service and database schema

* Make linter happy

* Linter ignore

* Rename datastore class

* Add basic CRUD for notifications, metadata, and a helper for logs

* Make linter happy again

* Shorten table names, suffix dates with gmt, introduce dedicated meta store

* Fix docblock on reusable datastore

* Lint fixes

* Lint fixes

* Update service provider

* Lint fixes

* Lint fixes

* Lint fixes

* Lint fixes

* Fix dbDelta issues with capital letters and phpcs fixes

* Add specific query formats

* Remove object keyword refs

* Remove meta cache duplication in favor of WC_Data level caching

* Improvements in stock notifications CRUD and introduce WP_Error returns

* Docblock improvements

* Create the main controller and wire the data store

* Fix data stores for basic CRUD and add unit tests

* Add metadata handling and unit tests

* Lint fixes

* Organize unit tests and align create and modified dates

* Alias main controller init and make linter happy

* One more lint fix

* Imporvements and new db table indexes

* Clean up notification unit tests

* Add metadata unit tests

* Rename the event log helper

* Add a factory, unit tests, and move notifications tests

* Lint fixes

* Make linter happy

* Add a query method

* Create data store for logs

* Lint fixes

* Lint fixes and fix the init data store of logs

* Fix error in log params

* Use the main datastore for tests

* Use specified date modified value and some polishing

* Fix unit tests by applying changes on the object after saving the meta

* Lint fixes

* Setup timezone for tests to control output

* Avoid error logging in unit tests

* Fix timezones when reading data from DB

* Lint fixes

* Reorg the table names into getters in data stores

* Minor fixes on the scheme

* Move delete functions in each data store

* Lint fixes

* Fix unit tests

* Remove comment ignores

* Use assertCount

* Fix unit tests

* Add args and return types in activity log data store

* Added args and return types in notifications data store where applicable

* Fix the datastore docblock in activity logs tests

* Differentiate dates for better coverage

* Remove text based tests

* Check non-existance after deleting meta but not saved yet

* One more rename of note to payload

* Introduce the email templates

* Simplify and reorg templates and controllers

* Lint fixes

* Add plain templates

* Handle plain mode in template functions

* Lint fixes

* Lint fixes

* Reorg of files, renamed the email classes, and introduced a single entrypoint on the Controller class

* Lint fixes

* Lint fixes

* revert unrelated changes

* Rename email titles

* Modify the database schema and fix the tests

* Make linter happy

* Add the id at the batch processing index

* Fix into->intro type

* MIC: Introduce Customer Stock Notifications settings

* Lint fixes

* Improve unit tests

* Group Guest sign-up options

* Lint fixes

* Rename custom tax location method

* Fix verify template

* Remove unused variables

* Remove unused argument

* Lint fixes

* Improve getter for formatted variations and add a unit test

* Add unit tests for get_product_permalink/name

* Rename emails id and classes for better semantics

* Lint fixes on tests

* Lint fixes on tests

* Reorg namespaces into Emails, and move the templates into the email controller

* Rename EmailManager, move two emails in transactional emails actions and modify trigger points

* Lint

* Fix customer data hook

* - Fix conditional visibility of Guest Sign-up options.
- Remove Minimum Stock Quantity option.
- Add extra validations for warnings

* Removed type hints from filter callbacks

* Update option names and enable sign ups for new stores

* Update option names in unit test

* Lint

* Lint

* Review

* Lint review

* More lint fixes

* More lint fixes

* Share the controller

* Few more conflicts

* One last conflict

* On more

* Oops

* Update option names and unit tests

* Lint fixes

* Review

---------

Co-authored-by: Chris Lilitsas <1105590+xristos3490@users.noreply.github.com>

* MIC: Add Customer Stock Notifications menus and Notifications List Tables (#58356)

* Add base service and database schema

* Make linter happy

* Linter ignore

* Rename datastore class

* Add basic CRUD for notifications, metadata, and a helper for logs

* Make linter happy again

* Shorten table names, suffix dates with gmt, introduce dedicated meta store

* Fix docblock on reusable datastore

* Lint fixes

* Lint fixes

* Update service provider

* Lint fixes

* Lint fixes

* Lint fixes

* Lint fixes

* Fix dbDelta issues with capital letters and phpcs fixes

* Add specific query formats

* Remove object keyword refs

* Remove meta cache duplication in favor of WC_Data level caching

* Improvements in stock notifications CRUD and introduce WP_Error returns

* Docblock improvements

* Create the main controller and wire the data store

* Fix data stores for basic CRUD and add unit tests

* Add metadata handling and unit tests

* Lint fixes

* Organize unit tests and align create and modified dates

* Alias main controller init and make linter happy

* One more lint fix

* Imporvements and new db table indexes

* Clean up notification unit tests

* Add metadata unit tests

* Rename the event log helper

* Add a factory, unit tests, and move notifications tests

* Lint fixes

* Make linter happy

* Add a query method

* Create data store for logs

* Lint fixes

* Lint fixes and fix the init data store of logs

* Fix error in log params

* Use the main datastore for tests

* Use specified date modified value and some polishing

* Fix unit tests by applying changes on the object after saving the meta

* Lint fixes

* Setup timezone for tests to control output

* Avoid error logging in unit tests

* Fix timezones when reading data from DB

* Lint fixes

* Reorg the table names into getters in data stores

* Minor fixes on the scheme

* Move delete functions in each data store

* Lint fixes

* Fix unit tests

* Remove comment ignores

* Use assertCount

* Fix unit tests

* Add args and return types in activity log data store

* Added args and return types in notifications data store where applicable

* Fix the datastore docblock in activity logs tests

* Differentiate dates for better coverage

* Remove text based tests

* Check non-existance after deleting meta but not saved yet

* One more rename of note to payload

* Introduce the email templates

* Simplify and reorg templates and controllers

* Lint fixes

* Add plain templates

* Handle plain mode in template functions

* Lint fixes

* Lint fixes

* Reorg of files, renamed the email classes, and introduced a single entrypoint on the Controller class

* Lint fixes

* Lint fixes

* revert unrelated changes

* Rename email titles

* Modify the database schema and fix the tests

* Make linter happy

* Add the id at the batch processing index

* Fix into->intro type

* MIC: Introduce Customer Stock Notifications settings

* Lint fixes

* Improve unit tests

* Group Guest sign-up options

* Lint fixes

* Rename custom tax location method

* Fix verify template

* Remove unused variables

* Remove unused argument

* Lint fixes

* Improve getter for formatted variations and add a unit test

* Add unit tests for get_product_permalink/name

* Rename emails id and classes for better semantics

* Lint fixes on tests

* Lint fixes on tests

* Reorg namespaces into Emails, and move the templates into the email controller

* Rename EmailManager, move two emails in transactional emails actions and modify trigger points

* Lint

* Fix customer data hook

* - Fix conditional visibility of Guest Sign-up options.
- Remove Minimum Stock Quantity option.
- Add extra validations for warnings

* Removed type hints from filter callbacks

* Update option names and enable sign ups for new stores

* Update option names in unit test

* Lint

* Lint

* Review

* Lint review

* More lint fixes

* More lint fixes

* Share the controller

* Initial commit to add controllers and Notifications List table

* Fix lint errors

* Few more conflicts

* One last conflict

* On more

* Add functionality to the Notifications list table

* Oops

* Lint fixes

* More lint fixes

* Fix function args

* Fix dependencies

* Update option names and unit tests

* Lint fixes

* Lint fixes

* Review

* Support filtering by month

* Lint fixes

* Support multiple statuses in query args

* Fix lint errors

* More lint fixes

* Handle bulk actions and deleting individual notifications

* Fix lint errors

* More lint fixes

* More lint fixes

* More lint fixes

* Remove inactive views, use pending/cancelled/sent instead

* Add JS script for confirmation when deleting a notification

* Lint fixes

* More lint fixes

* Lint and styling fixes

* Share the new classes in the Service Provider

* Support Order by functionality

* Lint fixes

* Improve blank state screen to look like Orders empty state screen

* Update screen ids

* More page slug updates

* Remove unused JS param

* More page url updates

* Fix SQL queries

* Review admin manager class

* Fix issues related to missing dates from the DB

* Updated logic around fetching notification months

* Add more context to translatable strings

* Move action handling to NotificationPage

* Fix lint errors

* Fill in missing creation date when updating

* Use a contractor instead of init and action instead of a section

* Lint fixes

* Handle missing creation date when updating

* Refactor the way notices are displayed

* Lint fixes

* Lint fixes

* Review

* Update folder name

* Review

* Delete plugins/woocommerce/src/Internal/StockNotifications/Admin/Views/html-admin-notifications.php

Remove template with capital V for views

* Cleanup templates

* Fix initial blankState

---------

Co-authored-by: Chris Lilitsas <1105590+xristos3490@users.noreply.github.com>

* MIC: Fix admin notices on stock notifications list table (#59006)

Fix list-table notices structure

* MIC: remove customer data from stock notification via privacy erasers (#59093)

Currently when a customer request data erasure, Woo will anonymize customer data on orders. After this PR, Woo will also anonymize customer data on stock notifications.

* MIC: Support manual notifications creation + edit (#59047)

* MIC: Support manual notifications creation

* Fix lint errors

* More lint fixes

* More lint fixes

* More lint fixes + added email validation

* Ignore to-do comment

* Maybe now I can keep my to-do comment?

* Support editing notifications

* Lint fixes

* More lint fixes

* More lint fixes

* Review

* Remove whitespace

* Introduce a new NotificationsPage::add_notice

* Fix lint errors

* Reorder action handlers for perf

* Add an alert when deleting permanently

* Handle gracefully non-existing users

* Minor typo fix

* review on list table and remove the waiting since column

* Support pre-filled values when errors occure in form and introduce specific validation during creation

* Lint fixes

* Minor fix on the add_notice helper

* Lint fixes

* Lint fixes

* Ensure notice gets cleaned up if invalid

* Rearrange form edit-handler

* Add more data model updates on single-edit actions and remove the verification email for now

* Rename views to Templates

* Avoid using static functions

* few more defensive parts on notices handler

* Few more defensive parts

* Revert list table change

* Remove unneeded helper on EmailManager

---------

Co-authored-by: Chris Lilitsas <1105590+xristos3490@users.noreply.github.com>

* MIC: Introduce stock tracking hooks and AS scheduling for Back in Stock Notifications (#58388)

* Add base service and database schema

* Make linter happy

* Linter ignore

* Rename datastore class

* Add basic CRUD for notifications, metadata, and a helper for logs

* Make linter happy again

* Shorten table names, suffix dates with gmt, introduce dedicated meta store

* Fix docblock on reusable datastore

* Lint fixes

* Lint fixes

* Update service provider

* Lint fixes

* Lint fixes

* Lint fixes

* Lint fixes

* Fix dbDelta issues with capital letters and phpcs fixes

* Add specific query formats

* Remove object keyword refs

* Remove meta cache duplication in favor of WC_Data level caching

* Improvements in stock notifications CRUD and introduce WP_Error returns

* Docblock improvements

* Create the main controller and wire the data store

* Fix data stores for basic CRUD and add unit tests

* Add metadata handling and unit tests

* Lint fixes

* Organize unit tests and align create and modified dates

* Alias main controller init and make linter happy

* One more lint fix

* Imporvements and new db table indexes

* Clean up notification unit tests

* Add metadata unit tests

* Rename the event log helper

* Add a factory, unit tests, and move notifications tests

* Lint fixes

* Make linter happy

* Add a query method

* Create data store for logs

* Lint fixes

* Lint fixes and fix the init data store of logs

* Fix error in log params

* Use the main datastore for tests

* Use specified date modified value and some polishing

* Fix unit tests by applying changes on the object after saving the meta

* Lint fixes

* Setup timezone for tests to control output

* Avoid error logging in unit tests

* Fix timezones when reading data from DB

* Lint fixes

* Reorg the table names into getters in data stores

* Minor fixes on the scheme

* Move delete functions in each data store

* Lint fixes

* Fix unit tests

* Remove comment ignores

* Use assertCount

* Fix unit tests

* Add args and return types in activity log data store

* Added args and return types in notifications data store where applicable

* Fix the datastore docblock in activity logs tests

* Differentiate dates for better coverage

* Remove text based tests

* Check non-existance after deleting meta but not saved yet

* One more rename of note to payload

* Introduce the email templates

* Simplify and reorg templates and controllers

* Lint fixes

* Add plain templates

* Handle plain mode in template functions

* Lint fixes

* Lint fixes

* Reorg of files, renamed the email classes, and introduced a single entrypoint on the Controller class

* Lint fixes

* Lint fixes

* revert unrelated changes

* Rename email titles

* Modify the database schema and fix the tests

* Make linter happy

* Add the id at the batch processing index

* Fix into->intro type

* Rename custom tax location method

* Fix verify template

* Improve getter for formatted variations and add a unit test

* Add unit tests for get_product_permalink/name

* Rename emails id and classes for better semantics

* Lint fixes on tests

* Lint fixes on tests

* Reorg namespaces into Emails, and move the templates into the email controller

* Rename EmailManager, move two emails in transactional emails actions and modify trigger points

* Lint

* Fix customer data hook

* Renamed confirm email to verified, and added better docblock for the customer data hook

* Simplify and repurpose get_product_formatted_variation_list; handle emails rendering on the spot

* Rename the template

* Lint

* Initial commit

* Refactor the notificationsProcessor and sync controller

* Introduce the stock sync and async processor controllers and tests

* Phpcs fixes

* More lint fixes

* More lint fixes

* More lint fixes

* More lint fixes

* Repurpose StockManagementHelper and cleanups

* Lint

* Prevent query for variations if no children

* Rename the helper for better semantics

* Fix AS args

* Fix AS args in tests

* Implement the attempt limit in the query

* Add unit tests for the attempt limit on query

* Cleanup debugs

* Make the unit test for skipped a bit more specific

* Fix parsing new AS args

* Final changes for the new AS args structure

* Use spread op

* Simplify throttle retun

* Specify array elements

* Rename and add DISTICT in query instead of array_unique

* Add runtime cache in config and rename hooks

* Add the EligibilityService

* Fix stocksynccontroller based on the new service

* Fix tests for stocksync and minor fix in notificationtests

* Use the aggregated hook in the processor

* Refactor logic into smaller components

* Lint fixes

* More lint fixes

* minor rearrangement

* Fix unit tests of the async processor

* Cleanup

* Add new tables in WC_Install::get_tables()

* Lint fixes

* Remove deprecated service provider

* Add typed class props on processor

* Polish sync hook

* Cache queue instance into prop

* Use local product_id ref

* harmonize logs group

* Oops

* Harmonize the logs group

* Properly handle unanavailable products

* Add another test for trashed products

* Fix the notice URL

* Customer Stock Notifications: daily cleanup task (#59331)

Adds a new setting under woo products so an admin can set the number of days to keep stale notifications. Runs a daily task to delete stale notifications.

Co-authored-by: roccotrip <rocco.tripaldi@automattic.com>

* Fix product link in the Back-in-Stock Notifications List Table (#59594)

Fix product link in the List Table

* [MiC] introduce cryptography tokens for handling unsubscribe for customer stock notifications (#59492)

* add new field to data model

* Add methods for getting email links

* move key to metadata part 1

* update option name for consistency

* add data methods

* back out of stale changeset

* add some doc blocks

* Default expiration time should be an hour

* simplify methods based on feedback from xristos3490

* linting

* Unit tests

* linting

* Default should be 1 hour

* fix linting for doc blocks

* fix broken link in contributing doc

* Runtime cache on Config and minor fixes

* Add a HashHelper util

* Implement the helper on the data model

---------

Co-authored-by: roccotrip <rocco.tripaldi@automattic.com>
Co-authored-by: Chris Lilitsas <1105590+xristos3490@users.noreply.github.com>

* MIC: Ensure that managed variations appear in the List Table when filtered by a product variable ID (#59595)

* Introduce the eligibility service as a DI on the list table and some cleanups

* Fetch the list table via the container

* Introduce Product Page Form for Back in Stock Notifications (#59470)

* Add the frontend template

* Add assets and scripts

* Add frontend script

* Extend jQuery and polish the script

* Lint

* Fix opt-in request data checking

* Imporve a11y on the frontend form

* Lint

* A few more a11y tweaks and fix the product-level setting

* Minor styling fix

* Fix nonce failures when global signups are off

* Decouple settings namespace from utils

* MIC: Process unsubscribe and verification email links (#59622)

* Add a controller for processing email actions

* process actions if key matches

* add a few unit tests

---------

Co-authored-by: roccotrip <rocco.tripaldi@automattic.com>

* Add constant gate on the main controller

* Add constant gate on the sql schema

* Skip unit tests

* Lint

* Bump versions from 0.0.0 to 10.2.0

* Changelog

* Bump template versions and fix plain filenames

* Bump the frontend template

* Fix the options DOM in the ListTable

* Avoid str_contains for php7.4 support

* User Jetpack constants for feature gating

* Bring back the tests

* Remove empty lines

* Use Jetpack Constants in all cases

---------

Co-authored-by: Jason Kytros <jason.kytros@automattic.com>
Co-authored-by: Rocco Tripaldi <tripaldir@gmail.com>
Co-authored-by: roccotrip <rocco.tripaldi@automattic.com>

tpaksu committed:
Add ambiguous provider match notice to tracking number lookup form (#60470)

* Add ambiguous provider match notice to tracking number lookup form

* Add changefile(s) from automation for the following project(s): woocommerce, woocommerce/client/admin

* Add tests

---------

Co-authored-by: github-actions <github-actions@github.com>

triple0t committed:
Fix: Email editor conflict with Site Editor and improve theme style inheritance (#60465)

* Refactor Email_Editor initialization to ensure assets manager is initialized on editor page.

This prevents it from loading on other pages i.e. the Site Editor

* Add isNestedEmpty utility function to check for empty nested objects in use-email-styles.ts

Enhance cleanupUserStyles function to remove keys with undefined values or empty nested objects. This ensures the editor does not save an empty object for the key preventing site-level style from being applied

e.g

instead of saving
```json
{"styles": {"color":[]}}
```

Save
```json
{"styles": {}}
```

* Ensure TypographyElementPanel reset typography styles to values froms Site editor or email editor theme.json

Using undefined here will force the remover of empty objects

Now, when the reset button is clicked, the style resets to the value from the site theme-json or email editor theme-json

* Add changelog files

github-actions[bot] committed:
Release: Remove 10.1.1 change files from trunk (#60492)

Delete changelog files from 10.1.1 release

Co-authored-by: WooCommerce Bot <no-reply@woocommerce.com>

albarin committed:
Update the `Bump version number` workflow to change the version (#60468)

* Update the bump version workflow to update the changelog date

* Use current year

* Update the version number

luisherranz committed:
iAPI Mini Cart: Hide the image when it doesn't load correctly (#60493)

* Hide the image when it doesn't load correctly

* Add changefile(s) from automation for the following project(s): woocommerce

* Add support for hidden from catalog products

* Remove empty line

---------

Co-authored-by: github-actions <github-actions@github.com>

naman03malhotra committed:
Move the resposibility of cred inputs from Registry to Platforms (#60420)

* remove unused mocks

* move the creds input configuration into the platform's registration

* Added changelog

* review feedback

github-actions[bot] committed:
Update stable tag to 10.1.1 (trunk) (#60505)

Update stable tag to 10.1.1 on trunk

Co-authored-by: github-actions <github-actions@github.com>

straku committed:
Fix padding for checkout steps when step numbers are enabled (#60473)

* Add new mixin to simplify the code

* Actual changes

* Add changelog

github-actions[bot] committed:
Prepare Packages for Release (#60519)

* Automated change: Prep @woocommerce/email-editor for release.

* Fix Markdown

---------

Co-authored-by: pavel-mailpoet <29194603+pavel-mailpoet@users.noreply.github.com>
Co-authored-by: Pavel Dohnal <pavel.dohnal@automattic.com>

github-actions[bot] committed:
Update changelog.txt after 10.1.1 (#60507)

Update changelog.txt after release 10.1.1

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

github-actions[bot] committed:
Prepare Packages for Release (#60521)

Automated change: Prep @woocommerce/email-editor-config for release.

Co-authored-by: pavel-mailpoet <29194603+pavel-mailpoet@users.noreply.github.com>

Aljullu committed:
Remove Add to Cart Button hardcoded size from Add to Cart + Options template parts (#60413)

* Remove Add to Cart Button hardcoded size from Add to Cart + Options template parts

* Add changelog file

* Styling fixes

* Fix styles in TT5 and TT3

NeosinneR committed:
More robust link detection in email preview (#60520)

* Add changelog

* More robust link detection in email preview

* Check for existence of DOMDocument

piinthecloud committed:
WIP: Move Extensibility docs from blocks folder to main docs folder (#60110)

* move files to docs and clean formatting

* add 'redirects' and fix duplicate content

* fix markdown errors

* fix markdown errors 2

* fix markdown errors 3

* fix markdown errors 4

* Add changefile(s) from automation for the following project(s): woocommerce

* address coderabbit improvements

* remove TOCs

---------

Co-authored-by: github-actions <github-actions@github.com>

SantosGuillamot committed:
Interactivity API powered MiniCart - Move MiniCart overlay to the `body` (#60464)

* Move minicart to the `body`

* Add changefile(s) from automation for the following project(s): woocommerce

* Fix linting

* Move logic to a class method

* Return `void` in method

---------

Co-authored-by: github-actions <github-actions@github.com>

dinhtungdu committed:
Migrate the product summary block to fetch product data via the `@wordpress/core-data` package (#60494)

Aljullu committed:
Fix a TypeScript error in addCartItem() (#60504)

* Fix TypeScript error in addCartItem()

* Add changelog file

* Fix wrong check

kmanijak committed:
Fix selector in E2E test to cover WP L-1 as well (#60528)

* Update selector to cover WP L-1

* Add changelog

sunyatasattva committed:
Bump opacity on sale price for legacy CSS to meet AAA-compliance across the board

This number ensures a contrast ratio of 7:1 with black text on white
background, ensuring AAA-compliance.

luisherranz committed:
iAPI Mini Cart: Pass an empty string when there's no short description (#60518)

* Pass an empty string when there's no short description

* Add changefile(s) from automation for the following project(s): woocommerce

* Avoid setting innerHTML when there are no descriptions

---------

Co-authored-by: github-actions <github-actions@github.com>

mordeth committed:
[NOX] Scroll to top of the page when navigating to Offline Gateways page (#60442)

* Scroll to top of the page when navigating to Offline Gateways page

mordeth committed:
[NOX] Refine RTL style handling for improved user experience (#60500)

* Refine RTL style handling for improved user experience

---------

Co-authored-by: Vlad Olaru <vlad@pixelgrade.com>
Co-authored-by: Vlad Olaru <vlad.olaru@automattic.com>

SantosGuillamot committed:
Interactivity API powered Mini Cart - Fix extra margin (#60537)

* Remove extra space

* Add changefile(s) from automation for the following project(s): woocommerce

---------

Co-authored-by: github-actions <github-actions@github.com>

mreishus committed:
Product Variations: Move post_excerpt updates from view time to write time (#59795)

NeosinneR committed:
Use store's locale when rendering email preview in settings (#60499)

* Use store's locale when rendering email preview in settings

* Add changelog

* User store's locale when generating dummy data for email preview

Mayisha committed:
create paypal helper class and helper method

Mayisha committed:
add notice stating the upcoming migration

Mayisha committed:
update copy

Mayisha committed:
show notice on the payments settings page

SantosGuillamot committed:
iAPI powered MiniCart - Fix variable product actions (#60497)

* Pass `variation` property in minicart actions

* Add changefile(s) from automation for the following project(s): woocommerce

* Add changefile(s) from automation for the following project(s): woocommerce

* Update variable name

---------

Co-authored-by: github-actions <github-actions@github.com>

yuliyan committed:
Add type check when extracting vars from theme styles in block emails (#60538)

* Add type check when extracting vars from theme style values in email editor

* Add changelog entry

ralucaStan committed:
Revert delayed initial rendering for checkout blocks (#60509)

* Revert delayed rendering for checkout blocks
This commit reverts the delay we add added https://github.com/woocommerce/woocommerce/pull/60020 because it broke the functionality of the interactivity api power product collection in the cart block.
The iAPI gets initialized before the inner blocks render, which causes the iapi to now work for blocks added with a delay

* Add changelog

Mayisha committed:
dismiss  notice

Mayisha committed:
don't show the notice if paypal is disabled

Mayisha committed:
Merge branch 'trunk' into task/paypal-tos-notice

Mayisha commented:
Closing this PR as I messed up with rebase and force push while moving it from a feature branch to trunk.
